starlite/exceptions.py

```diff
@@ -25,18 +27,40 @@ def __repr__(self) -> str:
 
 
 class MissingDependencyException(StarLiteException, ImportError):
-    pass
+    def __init__(self, feature: str, extras: tuple[str, ...]) -> None:
+        self.feature = feature
+        self.extras = extras
+
+        if len(extras) > 1:
+            _extras_repr = ", ".join(map(repr, extras[:-1])) + " or " + repr(extras[-1])
+        elif extras:
+            _extras_repr = repr(extras[0])
+        else:
+            raise RuntimeError(f"`extras` argument of {self.__class__.__name__} cannot be empty.")
+
+        detail = (
+            f"To use {feature}, install starlite with {_extras_repr} extra:\n"
+            f"e.g. `pip install starlite[{extras[0]}]`\n"
+            f"or `poetry add starlite --extras {extras[0]}`"
+        )
+        super().__init__(detail)
+
+        # this and custom Exception.__repr__ below is a workaround
+        # for https://github.com/starlite-api/starlite/issues/188
+        Exception.__init__(self, detail)
+
+    __repr__ = Exception.__repr__
 
 
 class HTTPException(StarLiteException, StarletteHTTPException):
     status_code = HTTP_500_INTERNAL_SERVER_ERROR
-    extra: Optional[Union[Dict[str, Any], List[Any]]] = None
+    extra: dict[str, Any] | list[Any] | None = None
```

Comment:
Syntax is not backwards compatible here, use `Union` instead throughout

Reply:
It is syntactically backwards compatible though: CI passes for every supported version.

I agree however to change it to Union to comply with the rest of the code base. 

Though I think it's worth to discuss perspective of upgrading annotations in the entire project somewhere.

Reply:
Interesting that it passes, but this is PEP 604--> https://peps.python.org/pep-0604/, which is python 3.10 only.

Reply:
It's the postponed annotations that make this correct though.. and these would be the only two modules in the whole project that introduce those.

> Though I think it's worth to discuss perspective of upgrading annotations in the entire project somewhere.

Yep, consistency is key.